iroh/src/endpoint.rs

```diff
@@ -919,6 +920,31 @@ impl Endpoint {
         self.msock.list_remote_infos().into_iter()
     }
 
+    /// Returns a stream of all remote nodes discovered through the endpoint's discovery services.
+    ///
+    /// Whenever a node is discovered via the endpoint's discovery service, the corresponding
+    /// [`DiscoveryItem`] is yielded from this stream. This includes nodes discovered actively
+    /// through [`Discovery::resolve`], which is invoked automatically when calling
+    /// [`Endpoint::connect`] for a [`NodeId`] unknown to the endpoint. It also includes
+    /// nodes that the endpoint discovers passively from discovery services that implement
+    /// [`Discovery::subscribe`], which e.g. [`LocalSwarmDiscovery`] does.
+    ///
+    /// The stream does not yield information about nodes that are added manually to the endpoint's
+    /// addressbook by calling [`Endpoint::add_node_addr`] or by supplying a full [`NodeAddr`] to
```

Comment:
What do you think should happen when `StaticProvider::add_node_addr` is called?  Should it show up here?

Reply:
Good question. I'm not sure?

Right now they would show up in the discovery stream only once you connect to a node added to the StaticProvider, but not at the time you add them to the StaticProvider.

Reply:
Added docs about current behavior. 

We could change this by having the `StaticProvider` impl `Discovery::subscribe` and emit a stream of all items added there, but that could be a followup (then it would also behave more like `add_node_addr` in that it adds the info the magic sock node map right away, not only on `connect` - not sure if that is desired?)